src/mc_tasks/ImpedanceTask.cpp

```diff
@@ -0,0 +1,311 @@
+/*
+ * Copyright 2015-2020 CNRS-UM LIRMM, CNRS-AIST JRL
+ */
+
+#include <mc_rbdyn/configuration_io.h>
+#include <mc_tasks/ImpedanceTask.h>
+#include <mc_tasks/MetaTaskLoader.h>
+
+#include <mc_rtc/gui/ArrayLabel.h>
+#include <mc_rtc/gui/Transform.h>
+
+namespace mc_tasks
+{
+
+namespace force
+{
+
+ImpedanceTask::ImpedanceTask(const std::string & surfaceName,
+                             const mc_rbdyn::Robots & robots,
+                             unsigned int robotIndex,
+                             double stiffness,
+                             double weight,
+                             double impM,
+                             double impD,
+                             double impK,
+                             double oriScale)
+: SurfaceTransformTask(surfaceName, robots, robotIndex, stiffness, weight), lowPass_(0.005, 0.05)
+{
+  const auto & robot = robots.robot(robotIndex);
+  type_ = "impedance";
+  name_ = "impedance_" + robots.robot(rIndex).name() + "_" + surfaceName;
+
+  if(!robot.surfaceHasIndirectForceSensor(surfaceName))
+  {
+    mc_rtc::log::error_and_throw<std::runtime_error>("[{}] Surface {} does not have a force sensor attached", name_,
+                                                     surfaceName);
+  }
+
+  // Set impedance parameters
+  impedance(sva::ForceVecd(Eigen::Vector3d::Constant(oriScale * impM), Eigen::Vector3d::Constant(impM)),
+            sva::ForceVecd(Eigen::Vector3d::Constant(oriScale * impD), Eigen::Vector3d::Constant(impD)),
+            sva::ForceVecd(Eigen::Vector3d::Constant(oriScale * impK), Eigen::Vector3d::Constant(impK)));
+}
+
+void ImpedanceTask::update(mc_solver::QPSolver & solver)
+{
+  // 1. Filter the measured wrench
+  if(lowPass_.dt() != solver.dt())
+  {
+    lowPass_.dt(solver.dt());
+  }
```

Comment:
You can override `addToSolver` to do this only once

Reply:
Thank you for telling nicer approach. Fixed in https://github.com/jrl-umi3218/mc_rtc/pull/100/commits/42ebb40896e82c93aca68902660b0c97032cb2ce.